central/networkpolicies/datastore/datastore_sac_test.go

```diff
@@ -0,0 +1,396 @@
+package store
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/store"
+	boltStore "github.com/stackrox/rox/central/networkpolicies/datastore/internal/store/bolt"
+	pgdbStore "github.com/stackrox/rox/central/networkpolicies/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/undodeploymentstore"
+	undodeploymentstoremock "github.com/stackrox/rox/central/networkpolicies/datastore/internal/undodeploymentstore/mocks"
+	"github.com/stackrox/rox/central/networkpolicies/datastore/internal/undostore"
+	undostoremock "github.com/stackrox/rox/central/networkpolicies/datastore/internal/undostore/mocks"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/bolthelper"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/testconsts"
+	"github.com/stackrox/rox/pkg/sac/testutils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+	bolt "go.etcd.io/bbolt"
+)
+
+func TestNetworkPolicySAC(t *testing.T) {
+	suite.Run(t, new(networkPolicySACSuite))
+}
+
+type networkPolicySACSuite struct {
+	suite.Suite
+
+	datastore DataStore
+
+	pool *pgxpool.Pool
+
+	engine *bolt.DB
+
+	storage             store.Store
+	undoStore           undostore.UndoStore
+	undoDeploymentStore undodeploymentstore.UndoDeploymentStore
+
+	mockCtrl *gomock.Controller
+
+	testContexts         map[string]context.Context
+	testNetworkPolicyIDs []string
+}
+
+func (s *networkPolicySACSuite) SetupSuite() {
+	var err error
+	if features.PostgresDatastore.Enabled() {
+		ctx := context.Background()
+		src := pgtest.GetConnectionString(s.T())
+		cfg, err := pgxpool.ParseConfig(src)
+		s.NoError(err)
```

Comment:
Let's use `s.Require().NoError(err)`, otherwise `t.Fail()` won't be called and we have potential subsequent errors later on (i.e. in this case a nil pointer in `pgxpool.ConnectConfig()`.

This applies to all occurrences of `s.NoError()` within `SetupSuite, TearDownSuite, deleteNetworkPolicy` and all test functions where some pre-reqs will be done (i.e. creating a network policy and upserting it).